src/ring_theory/hahn_series.lean

```diff
@@ -828,11 +833,11 @@ emb_domain_single.trans $ hf.symm ▸ rfl
 @[simps] def emb_domain_ring_hom (f : Γ →+ Γ') (hfi : function.injective f)
   (hf : ∀ g g' : Γ, f g ≤ f g' ↔ g ≤ g') :
   hahn_series Γ R →+* hahn_series Γ' R :=
-{ to_fun := emb_domain_linear_map ⟨⟨f, hfi⟩, hf⟩,
```

Comment:
There is no `emb_domain_add_monoid_hom`, right? I guess we shouldn't add it if we don't need it. (And it doesn't really fall in the scope of this PR.)

Reply:
I was thinking it should probably exist too, if only to provide `emb_domain_sub` and `emb_domain_neg` lemmas. I'd prefer to declare it out of scope though, since really this is just cleanup prework before generalizing in #6786 as the linear_map doesn't exist when `R` is non-associative.